From 7d7c9f1f4e170496eccb9cb1edac32736c11d0c4 Mon Sep 17 00:00:00 2001
From: Shwetha G K <quic_kshwetha@quicinc.com>
Date: Wed, 25 Sep 2024 20:49:53 +0530
Subject: [PATCH] cfg80211: Fix the STA MLD becoming dormat

Fix the case where STA MLD is getting dormant due to
link vifs getting reverted as part of dev_close(mldX).

Change-Id: I738feb250955b89cf770cffb980dcc51002f5f25
Signed-off-by: Shwetha G K <quic_kshwetha@quicinc.com>
---
 net/wireless/util.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/net/wireless/util.c b/net/wireless/util.c
index 1acead7457a9..11c2e5e0a622 100644
--- a/net/wireless/util.c
+++ b/net/wireless/util.c
@@ -2690,10 +2690,9 @@ void cfg80211_remove_links(struct wireless_dev *wdev)
 
 	/*
 	 * links are controlled by upper layers (userspace/cfg)
-	 * only for AP and STA mode, so only remove them here for AP or STA
+	 * only for AP, so only remove them here for AP
 	 */
-	if (wdev->iftype != NL80211_IFTYPE_AP &&
-		wdev->iftype != NL80211_IFTYPE_STATION)
+	if (wdev->iftype != NL80211_IFTYPE_AP)
 		return;
 
 	wdev_lock(wdev);
-- 
2.34.1

